### PR TITLE
fix: make azure dns resource group optional

### LINKED ISSUE #2342: Update docs with optional Azure DNS group

### PR DESCRIPTION
**Pull request by mrsimonemms**
## Description

Updated the 2.7 docs too

## Related Issue(s)

Fixes https://github.com/konstructio/kubefirst/issues/2342

### CODE REVIEW

File: docs/azure/partials/common/_dns-resource-group.mdx [hunk 1]
```diff
@@ -1 +1 @@
-The `dns-azure-resource-group` is only required if using Azure as your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned.
+The `dns-azure-resource-group` flag is optional and only if using Azure are your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned.
```
Comment (konstructcarrie) [inline docs/azure/partials/common/_dns-resource-group.mdx]: I'd rephrase to say (it's also not clear to me if it is optional with Azure why you'd use it?) 

"The `dns-azure-resource-group` flag is optional and only applies if Azure is your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned."

_Is it required or optional if Azure is your DNS provider, if using Azure makes it no longer optional I'd say_

"The `dns-azure-resource-group` flag is only required if Azure is your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned."

Comment (mrsimonemms) [reply docs/azure/partials/common/_dns-resource-group.mdx]: Is this better?

The flag is only used if Azure is the DNS provider (ie, ignored for Cloudflare). It's an optional flag even if Azure is used. See `resource_group_name` in the [Terraform docs](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/dns_zone#argument-reference) as this is doing the same thing (it's actually the inspiration for the functionality)

File: docs/azure/partials/common/_dns-resource-group.mdx [hunk 2]
```diff
@@ -1 +1 @@
-The `dns-azure-resource-group` is only required if using Azure as your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned.
+The `dns-azure-resource-group` flag is only used Azure is your DNS provider. This is the name of the resource group where the [DNS Zone](https://learn.microsoft.com/en-us/azure/dns/dns-zones-records) resource is provisioned. This flag is not required, but may be useful if you have multiple DNS Zones with the same URL configured. If the resource group is not provided, the first DNS Zone that matches the given URL in your subscription will be used.
```
Comment (konstructcarrie) [inline docs/azure/partials/common/_dns-resource-group.mdx]: Update that first bit to "The `dns-azure-resource-group` flag is only used if Azure is your DNS provider..." and we're good to go

**Review by konstructcarrie — APPROVED** [review]
👍
